hazelcast/src/main/java/com/hazelcast/internal/monitors/PerformanceLogFile.java

```diff
@@ -0,0 +1,301 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.monitors;
+
+import com.hazelcast.instance.BuildInfo;
+import com.hazelcast.instance.BuildInfoProvider;
+import com.hazelcast.instance.GroupProperties;
+import com.hazelcast.instance.HazelcastInstanceImpl;
+import com.hazelcast.instance.Node;
+import com.hazelcast.internal.management.dto.SlowOperationDTO;
+import com.hazelcast.internal.metrics.MetricsRegistry;
+import com.hazelcast.internal.metrics.renderers.CommaSeparatedKeyValueProbeRenderer;
+import com.hazelcast.internal.metrics.renderers.HumanFriendlyProbeRenderer;
+import com.hazelcast.internal.metrics.renderers.ProbeRenderer;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.spi.impl.operationservice.InternalOperationService;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Properties;
+
+import static com.hazelcast.nio.IOUtil.closeResource;
+import static java.lang.Math.round;
+import static java.lang.String.format;
+import static java.lang.System.currentTimeMillis;
+import static java.lang.System.getProperties;
+import static java.lang.System.getProperty;
+import static java.util.Collections.sort;
+
+final class PerformanceLogFile {
+
+    private static final int ONE_MB = 1024 * 1024;
+
+    volatile File logFile;
+
+    private final MetricsRegistry metricRegistry;
+    private final HazelcastInstanceImpl hazelcastInstance;
+    private final InternalOperationService operationService;
+    private final ILogger logger;
+
+    private final HeadRenderer headRenderer = new HeadRenderer();
+    private final BodyRendered bodyRendered;
+    private final String pathname;
+    private long lastModified;
+    private int index;
+    private BufferedWriter writer;
+    private boolean renderHead;
+    private int maxRollingFileCount;
+    private int maxRollingFileSizeBytes;
+
+    PerformanceLogFile(PerformanceMonitor performanceMonitor) {
+        this.bodyRendered = new BodyRendered(performanceMonitor.humanFriendlyFormat);
+        this.logger = performanceMonitor.logger;
+        this.hazelcastInstance = performanceMonitor.hazelcastInstance;
+        Node node = hazelcastInstance.node;
+        this.metricRegistry = node.nodeEngine.getMetricsRegistry();
+        this.operationService = node.nodeEngine.getOperationService();
+        this.pathname = "performance-" + hazelcastInstance.getName() + "-" + currentTimeMillis() + "-%03d.log";
```

Comment:
What about using member address (host + port) instead of `hazelcastInstance.getName()`? Because instance name probably will be the same for all members started in a separate process/machine. When you gather/pull all logs to a single directory, you cannot tell which log belongs to which member.